baybe/parameters/substance.py

```diff
@@ -156,19 +153,30 @@ def values(self) -> tuple:
         # for Python 3.7 or higher
         return tuple(self.data.keys())
 
+    # TODO: @Roya not sure if dynamic import in cached_property is a good practice
     @cached_property
     def comp_df(self) -> pd.DataFrame:  # noqa: D102
         # See base class.
         vals = list(self.data.values())
         pref = self.name + "_"
 
+        smiles_to_mordred_features = import_optional_module(
+            "baybe.utils.chemistry", "smiles_to_mordred_features", error="warn"
+        )
+        smiles_to_rdkit_features = import_optional_module(
+            "baybe.utils.chemistry", "smiles_to_rdkit_features", error="warn"
+        )
+        smiles_to_fp_features = import_optional_module(
+            "baybe.utils.chemistry", "smiles_to_fp_features", error="warn"
+        )
+
         # Get the raw descriptors
         if self.encoding is SubstanceEncoding.MORDRED:
-            comp_df = smiles_to_mordred_features(vals, prefix=pref)
+            comp_df = smiles_to_mordred_features(vals, prefix=pref)  # type: ignore
```

Comment:
what type error was thrown? would prefer to fix instead of ignore (unless of course not possible due to eg missing stubs or so)

Reply:
https://github.com/python/mypy/issues/5059#issuecomment-389455970

AFAIK it's the only way I could find to fix this problem.

Reply:
one way to do it would be to change the return type of `import_optional_module` in `optional.py` to `Any` instead of `Optional[Any]`

seems OK to me as the situations where return of `import_optional_module` is `None` should not really be handled by the type checker but are handled by whatever is specified in `error` argument. Id prefer that over `# type:` exceptions